### PR TITLE
Taxonomies: Add Hierarchical Input UI to the inspector 

### PR DESCRIPTION
**Pull request by samwhale**
## Context

Taxonomies. [Designs](https://www.figma.com/file/1YM3jA9h4QGc4xLdyaXIJS/Stories-Sprint-1.7?node-id=9721%3A383514)

## Summary

Adds the UI for categories to the taxonomies section. Nothing is hooked up to the backend yet.

## Relevant Technical Choices

Waiting on #9004 to hook it all up, so this PR is purely presentational.

## To-do

n/a

## User-facing changes

categories

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Enable the taxonomies feature flag
2. Open a story
3. Click on the `Document` tab in the inspector
4. Mess with the new UI - should be accessible and fun to play with

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

no

### Does this PR have a legal-related impact?

no

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Partially addresses #8841

### CODE REVIEW

File: packages/story-editor/src/components/panels/document/taxonomies/taxonomies.js [hunk 1]
```diff
@@ -18,18 +18,146 @@
...
+  label,
+  * > label {
+    ${({ theme }) =>
+      themeHelpers.expandPresetStyles({
+        preset:
+          theme.typography.presets.label[
+            THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL
+          ],
+        theme,
+      })};
```
Comment (samwhale) [inline packages/story-editor/src/components/panels/document/taxonomies/taxonomies.js:53]: The labels that are a part of the `Input` component have a `primary` color and a font size that is bigger than what's specified in the designs. This catches all labels in this section and makes them pretty.

File: packages/story-editor/src/components/panels/document/taxonomies/taxonomies.js [hunk 2]
```diff
@@ -18,18 +18,146 @@
...
+
   return (
     <SimplePanel
       name="taxonomies"
       title={__('Categories and Tags', 'web-stories')}
       {...props}
-    />
+    >
+      <ContentArea>
+        <ContentHeading>{__('Categories', 'web-stories')}</ContentHeading>
```
Comment (maxyinger) [inline packages/story-editor/src/components/panels/document/taxonomies/taxonomies.js:128]: awesome. thanks! will incorporate this into the tags panel when that gets merged.

**Review by maxyinger — APPROVED** [review]
looking fantastic. should integrate nicely with provider once it's in here

File: packages/story-editor/src/components/panels/document/taxonomies/taxonomies.js [hunk 3]
```diff
@@ -18,18 +18,146 @@
...
+  Text,
+  themeHelpers,
+  THEME_CONSTANTS,
+} from '@web-stories-wp/design-system';
+import { useCallback, useMemo, useState } from '@web-stories-wp/react';
+import { v4 as uuidv4 } from 'uuid';
 /**
  * Internal dependencies
  */
+import styled from 'styled-components';
```
Comment (BrittanyIRL) [inline packages/story-editor/src/components/panels/document/taxonomies/taxonomies.js]: nit: should be with external

**Review by BrittanyIRL — APPROVED** [review]
🎉  Awesome! Thanks for separating the UI and the logic for review.

**Pull request merged**